dev_guide/image_signatures.adoc

```diff
@@ -0,0 +1,192 @@
+
+----
+PUT /extensions/v2/<namespace>/<name>/signatures/<digest>
+----
+
+----
+$ curl -X PUT --data @signature.json http://<user>:<token>@<registry_endpoint>:5000/extensions/v2/<namespace>/<name>/signatures/sha256:<digest>
+----
+
+The JSON payload with the signature content should have the following structure:
+
+----
+{
+  "version": 2,
+  "type":    "atomic",
+  "name":    "sha256:4028782c08eae4a8c9a28bf661c0a8d1c2fc8e19dbaae2b018b21011197e1484@cddeb7006d914716e2728000746a0b23",
+  "content": "<base64_encoded_signature>"
+}
+----
+
+The `name` field contains the name of the image signature, which must be unique
+and in the format `<digest>@<name>`. The `<digest>` represents an image name and
+the `<name>` is the name of the signature. The signature name must be 32
+characters long.
+
+[NOTE]
+====
+In order to attach the signature to the image, the user must have the
+`system:image-signer` role in the relevant project.
+====
+
+[[reading-image-signatures-via-registry-api]]
+=== Reading Image Signatures
+
+Assuming a signed image has already been pushed into the OpenShift Container
+Registry, you can read the signatures using the following command:
+
+----
+GET /extensions/v2/<namespace>/<name>/signatures/<digest>
+----
+
+----
+$ curl http://<user>:<token>@<registry_endpoint>:5000/extensions/v2/<namespace>/<name>/signatures/sha256:<digest>
+----
+
+The `<namespace>` represents the {product-title} project name or registry
+repository name and the `<name>` refers to the name of the image repository. The
+`digest` represents the SHA-256 checksum of the image.
+
+If the given image contains the signature data, the output of the command above
+should produce following JSON response:
+
+----
+{
+  "signatures": [
+  {
+    "version": 2,
+    "type":    "atomic",
+    "name":    "sha256:4028782c08eae4a8c9a28bf661c0a8d1c2fc8e19dbaae2b018b21011197e1484@cddeb7006d914716e2728000746a0b23",
+    "content": "<base64_encoded_signature>"
+  }
+  ]
+}
+----
+
+The `name` field contains the name of the image signature, which must be unique
+and in the format `<digest>@<name>`. The `<digest>` represents an image name and
+the `<name>` is the name of the signature. The signature name must be 32
+characters long.
+
+The `content` contains Base64-encoded signature data (usually GPG-signed data).
+
+[[verifying-image-signatures-using-openshift-cli]]
+== Verifying Image Signatures Using OpenShift CLI
+
+You can verify the signatures of an image imported to an OpenShift Container
+Registry using the `oc adm verify-image-signature` command. This command
+verifies if the image identity contained in the image signature can be trusted
+by using the public GPG key to verify the signature itself then match the
+provided expected identity with the identity (the pull spec) of the given image.
+
+By default, this command uses the public GPG keyring located in
+*_$GNUPGHOME/.gnupg/pubring.gpg_*. By default, this command does not save the
```

Comment:
*$GNUPGHOME/pubring.gpg*, typically in path *~/.gnupg*.

See https://github.com/openshift/origin/pull/15080